File: electric.py
class customersDC:
    def __init__(self):
        self.customerNAME = ""
        self.classiFICT = ""
        self.totalKWU = 0.0
        self.ratePKU = 0.0
        self.groAMT = 0.0
        self.misceCHAR = 0.0
        self.totalAMT = 0.0
        self.prevREADING = 0.0
        self.currREADING = 0.0
        self.meterID = 0

customersD = []

def main():
    while True:
        print("\n[ELECTRIC COMPANY]")
        print("1. Calculate Bill")
        print("2. Visit Database")
        print("3. Exit")
        try:
            choice = int(input("Enter: "))
        except ValueError:
            choice = -1
        match choice:
            case 1:
                elecCAL()
            case 2:
                dataBASE()
            case 3:
                exit()
                return
            case _:
                for i in range(1, 4):
                    print("\n[PLEASE ENTER A VALID INPUT]")

def elecCAL():
    customers = customersDC()
    print("\n[CALCULATE ELECTRIC BILL]")
    customers.customerNAME = input("Enter Customer Name: ")
    customers.meterID = int(input("\nEnter Meter ID: "))
    customers.classiFICT = input("\nEnter Classification: ")
    customers.prevREADING = float(input("\nEnter Previous Reading: "))
    customers.currREADING = float(input("\nEnter Current Reading: "))
    #bruh
    customers.totalKWU = customers.currREADING - customers.prevREADING
    if customers.classiFICT.lower() == "residential":
        if customers.totalKWU > 300:
            customers.misceCHAR = 0.8
            customers.ratePKU = 0.55
            customers.groAMT = customers.totalKWU * customers.ratePKU
            customers.totalAMT = customers.groAMT + customers.misceCHAR
            customersD.append(customers)
            SAVEDFILES()
            disPLAY()
        elif customers.totalKWU >= 100 and customers.totalKWU <= 299:
            customers.misceCHAR = 0.5
            customers.ratePKU = 0.55
            customers.groAMT = customers.totalKWU * customers.ratePKU
            customers.totalAMT = customers.groAMT + customers.misceCHAR
            customersD.append(customers)
            SAVEDFILES()
            disPLAY()
        elif customers.totalKWU < 100:
            customers.misceCHAR = 0
            customers.ratePKU = 0.55
            customers.groAMT = customers.totalKWU * customers.ratePKU
            customers.totalAMT = customers.groAMT + customers.misceCHAR
            customersD.append(customers)
            SAVEDFILES()
            disPLAY()
    elif customers.classiFICT.lower() == "commercial":
        if customers.totalKWU > 300:
            customers.misceCHAR = 0.8
            customers.ratePKU = 0.75
            customers.groAMT = customers.totalKWU * customers.ratePKU
            customers.totalAMT = customers.groAMT + customers.misceCHAR
            customersD.append(customers)
            SAVEDFILES()
            disPLAY()
        elif customers.totalKWU >= 100 and customers.totalKWU <= 299:
            customers.misceCHAR = 0.5
            customers.ratePKU = 0.75
            customers.groAMT = customers.totalKWU * customers.ratePKU
            customers.totalAMT = customers.groAMT + customers.misceCHAR
            customersD.append(customers)
            SAVEDFILES()
            disPLAY()
        elif customers.totalKWU < 100:
            customers.misceCHAR = 0
            customers.ratePKU = 0.75
            customers.groAMT = customers.totalKWU * customers.ratePKU
            customers.totalAMT = customers.groAMT + customers.misceCHAR
            customersD.append(customers)
            SAVEDFILES()
            disPLAY()

def dataBASE():
    try:
        print("\n[CUSTOMERS' DATABASE]")
        with open("customersPY.txt", "r") as reader:
            for line in reader:
                print(line.strip())
    except IOError:
        for _ in range(3):
            print("\n[ERROR]")

def disPLAY():
    for customers in customersD:
        print(f"Customer Name: {customers.customerNAME}")
        print(f"Total Kilowatt Usage: {customers.totalKWU:.2f} kWh")
        print(f"Rate Per Kilowatt: {customers.ratePKU:.2f} kWh")
        print(f"Gross Amount: {customers.groAMT:.2f} PHP")
        print(f"Miscellaneous Charges: {customers.misceCHAR:.2f} PHP")
        print(f"Total Amount Due: {customers.totalAMT:.2f} PHP")

def exit():
    print("\n[THANK YOU FOR USING!]")

def SAVEDFILES():
    try:
        with open("customersPY.txt", "w") as writer:
            for customers in customersD:
                writer.write(f"CUSTOMER NAME: {customers.customerNAME}\n")
                writer.write(f"METER ID: {customers.meterID}\n")
                writer.write(f"PREVIOUS READING: {customers.prevREADING:.2f} kWh\n")
                writer.write(f"CURRENT READING: {customers.currREADING:.2f} kWh\n")
                writer.write(f"TOTAL KILOWATT USAGE: {customers.totalKWU:.2f} kWh\n")
                writer.write(f"RATE PER KILOWATT: {customers.ratePKU:.2f} kWh\n")
                writer.write(f"GROSS AMOUNT: {customers.groAMT:.2f} PHP\n")
                writer.write(f"MISCELLANEOUS CHARGES: {customers.misceCHAR:.1f} PHP\n")
                writer.write(f"TOTAL AMOUNT DUE: {customers.totalAMT:.2f} PHP\n")
    except IOError:
        for _ in range(3):
            print("\n[ERROR]")

if __name__ == "__main__":
    main()

File: electric.java
import java.util.*;
import java.io.*;

class customersDC {
    String customerNAME, classiFICT;
    double totalKWU, ratePKU, groAMT, misceCHAR, totalAMT, prevREADING, currREADING;
    int meterID;
}

public class electric {
    static Scanner input = new Scanner(System.in);
    static List<customersDC> customersD = new ArrayList<>();
    public static void main(String[] args) {
        while (true) {
            int choice;
            System.out.println("\n[ELECTRIC COMPANY]");
            System.out.print("\n1. Calculate Bill");
            System.out.print("\n2. Visit Database");
            System.out.println("\n3. Exit");
            System.out.print("\nEnter: ");
            try {
                choice = Integer.parseInt(input.nextLine());
            } catch (NumberFormatException e) {
                choice = -1;
            }
            switch (choice) {
                case 1:
                elecCAL();
                break;
                case 2:
                dataBASE();
                break;
                case 3:
                exit();
                return;
                default:
                for (int i = 0; i < 3; i++) {
                    System.out.println("\n[PLEASE ENTER A VALID INPUT]");
                }
            }
        }
    }

    public static void elecCAL() {
        customersDC customers = new customersDC();
        System.out.println("\n[CALCULATE ELECTRIC BILL]");
        System.out.print("\nEnter Customer Name: ");
        customers.customerNAME = input.nextLine();
        System.out.print("\nEnter Meter ID: ");
        customers.meterID = input.nextInt();
        input.nextLine();
        System.out.print("\nEnter Classification: ");
        customers.classiFICT = input.nextLine();
        System.out.print("\nEnter Previous Reading: ");
        customers.prevREADING = input.nextDouble();
        System.out.print("\nEnter Current Reading: ");
        customers.currREADING = input.nextDouble();
        input.nextLine();
        //BRUH
        customers.totalKWU = customers.currREADING - customers.prevREADING;
        if (customers.classiFICT.equalsIgnoreCase("residential")) {
            if (customers.totalKWU > 300) {
                customers.misceCHAR = 0.8;
                customers.ratePKU = 0.55;
                customers.groAMT = customers.totalKWU * customers.ratePKU;
                customers.totalAMT = customers.groAMT + customers.misceCHAR;
                customersD.add(customers);
                SAVEDFILES();
                disPLAY();
            } else if (customers.totalKWU >= 100 && customers.totalKWU <= 299) {
                customers.misceCHAR = 0.5;
                customers.ratePKU = 0.55;
                customers.groAMT = customers.totalKWU * customers.ratePKU;
                customers.totalAMT = customers.groAMT + customers.misceCHAR;
                customersD.add(customers);
                SAVEDFILES();
                disPLAY();
            } else if (customers.totalKWU < 100) {
                customers.misceCHAR = 0;
                customers.ratePKU = 0.55;
                customers.groAMT = customers.totalKWU * customers.ratePKU;
                customers.totalAMT = customers.groAMT + customers.misceCHAR;
                customersD.add(customers);
                SAVEDFILES();
                disPLAY();
            }
        } else if (customers.classiFICT.equalsIgnoreCase("commercial")) {
            if (customers.totalKWU > 300) {
                customers.misceCHAR = 0.8;
                customers.ratePKU = 0.75;
                customers.groAMT = customers.totalKWU * customers.ratePKU;
                customers.totalAMT = customers.groAMT + customers.misceCHAR;
                customersD.add(customers);
                SAVEDFILES();
                disPLAY();
            } else if (customers.totalKWU >= 100 && customers.totalKWU <= 299) {
                customers.misceCHAR = 0.5;
                customers.ratePKU = 0.75;
                customers.groAMT = customers.totalKWU * customers.ratePKU;
                customers.totalAMT = customers.groAMT + customers.misceCHAR;
                customersD.add(customers);
                SAVEDFILES();
                disPLAY();
            } else if (customers.totalKWU < 100) {
                customers.misceCHAR = 0;
                customers.ratePKU = 0.75;
                customers.groAMT = customers.totalKWU * customers.ratePKU;
                customers.totalAMT = customers.groAMT + customers.misceCHAR;
                customersD.add(customers);
                SAVEDFILES();
                disPLAY();
            }
        }
    }

    public static void dataBASE() {
        try (BufferedReader reader = new BufferedReader(new FileReader("customers.txt"))) {
            String line;
            System.out.println("\n[CUSTOMERS' DATABASE]");
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            for (int i = 0; i < 3; i++) {
                System.out.println("\nERROR");
            }
        }
    }

    public static void disPLAY() {
        for (customersDC customers : customersD) {
            System.out.println("Customer Name: " + customers.customerNAME);
            System.out.printf("Total Kilowatt Usage: %.2f kWh%n", customers.totalKWU);
            System.out.printf("Rate Per Kilowatt: %.2f kWh%n", customers.ratePKU);
            System.out.printf("Gross Amount: %.2f PHP%n", customers.groAMT);
            System.out.printf("Miscellaneous Charges: %.1f PHP%n", customers.misceCHAR);
            System.out.printf("Total Amount Due: %.2f PHP%n", customers.totalAMT);
        }
    }

    public static void exit() {
        System.out.println("\n[THANK YOU FOR USING!]");
    }

    public static void SAVEDFILES() {
        try (PrintWriter writer = new PrintWriter("customers.txt")) {
            for (customersDC customers : customersD) {
                writer.println("CUSTOMER NAME: " + customers.customerNAME);
                writer.println("METER ID: " + customers.meterID);
                writer.printf("PREVIOUS READING: %.2f kWh%n", customers.prevREADING);
                writer.printf("CURRENT READING: %.2f kWh%n", customers.currREADING);
                writer.printf("TOTAL KILOWATT USAGE: %.2f kWh%n", customers.totalKWU);
                writer.printf("RATE PER KILOWATT: %.2f kWh%n", customers.ratePKU);
                writer.printf("GROSS AMOUNT: %.2f PHP%n", customers.groAMT);
                writer.printf("MISCELLANEOUS CHARGES: %.1f PHP%n", customers.misceCHAR);
                writer.printf("TOTAL AMOUNT DUE: %.2f PHP%n", customers.totalAMT);
            }
        } catch (IOException e) {
            for (int i = 0; i < 3; i++) {
                System.out.println("[ERROR]");
            }
        }
    }
}